lib/_stream_readable.js

```diff
@@ -1132,7 +1132,7 @@ async function* createAsyncIterator(stream) {
 
   try {
     while (true) {
-      const chunk = stream.read();
+      const chunk = stream.destroyed ? null : stream.read();
```

Comment:
@mcollina: This kind of makes sense, but is different from master (should probably include it there as well?)

@lpinca ping as well

Reply:
Hmm taking out the data after the stream is destroyed via `stream.read()` is a must. I don't know if this is acceptable in the async iterator case.

Reply:
> Hmm taking out the data after the stream is destroyed via stream.read() is a must. 

That will still work.

Reply:
This PR is actually *less* "breaking" than the original PR.

Reply:
Yes but my point is that when consuming the stream with `for await...of` some data might not be read. If that happens also when using a `'data'` listener then it's ok.

Reply:
Ok, I have no objections.